open-api/rest-catalog-open-api.yaml

```diff
@@ -3964,6 +4144,292 @@ components:
         metadata:
           $ref: '#/components/schemas/TableMetadata'
 
+    EventsResponse:
+      type: object
+      required:
+        - highest-processed-timestamp-ms
+        - events
+      properties:
+        next-page-token:
+          $ref: "#/components/schemas/PageToken"
+        highest-processed-timestamp-ms:
+          description: >
+            The highest event timestamp processed when generating this response. 
+            This may not necessarily appear in the returned changes if it was filtered out.
+          type: integer
+          format: int64
+        events:
+          type: array
+          items:
+            $ref: "#/components/schemas/Event"
+
+    Event:
+      type: object
+      required:
+        - event-id
+        - request-id
+        - event-count
+        - timestamp-ms
+        - operation
+      properties:
+        event-id:
+          type: string
+          description: Unique ID of this event. Clients should perform deduplication based on this ID.
+        request-id:
+          description: >
+            Opaque ID of the request this change belongs to.
+            This ID can be used to identify events that were part of the same request.
+            Servers generate this ID randomly.
+          type: string
+        event-count:
+          type: integer
+          description: Number of events in the request / batch of events
```

Comment:
Is this event made up of one or more other events? If not, why is event count within event? Is it an ordinal?

It isn't clear what this is for or how to produce it from just looking at the spec, but it is required.

Reply:
New description:

```md
Total number of events in this batch or request.
Some endpoints, such as "updateTable" and "commitTransaction", can perform multiple updates in a single atomic request.
Each update is modeled as a separate event. All events generated by the same request share the same `request-id`.
The `event-count` field indicates the total number of events generated by that request.
```